### ISSUE TITLE
Build fails on macOS 10.13 with Xcode 9.2

### ISSUE THREAD

**Issue by apjanke** [SuperTux/supertux#762]
**SuperTux version: 0.5.1**
**System information:** macOS 10.3.3, Xcode 9.2

I can't tell if it also happens on the head of `master`; I can't get that to build, for apparently unrelated reasons.

##### Expected behavior

`cmake -DPNG_PNG_INCLUDE_DIR=/usr/local/opt/libpng; make` compiles SuperTux successfully.

##### Actual behavior

Compilation error:

```
...
In file included from /Users/janke/tmp/build-supertux/SuperTux-v0.5.1-Source/external/squirrel/squirrel/sqvm.cpp:5:
In file included from /Applications/Xcode-9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/math.h:310:
/Applications/Xcode-9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/limits:147:85: error:
      expected expression
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type min() _NOEXCEPT {return type();}
                                                                                    ^
/Users/janke/tmp/build-supertux/SuperTux-v0.5.1-Source/external/squirrel/squirrel/sqobject.h:131:24: note: expanded from macro
      'type'
#define type(obj) ((obj)._type)
                       ^
In file included from /Users/janke/tmp/build-supertux/SuperTux-v0.5.1-Source/external/squirrel/squirrel/sqvm.cpp:5:
In file included from /Applications/Xcode-9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/math.h:310:
/Applications/Xcode-9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/limits:148:85: error:
      expected expression
    _LIBCPP_INLINE_VISIBILITY static _LIBCPP_CONSTEXPR type max() _NOEXCEPT {return type();}
                                                                                    ^
/Users/janke/tmp/build-supertux/SuperTux-v0.5.1-Source/external/squirrel/squirrel/sqobject.h:131:24: note: expanded from macro
      'type'
#define type(obj) ((obj)._type)
                       ^
...
```

Full build log: https://gist.github.com/apjanke/efb08aa3c3a5691a45f0339d0796a759

##### Steps to reproduce actual behavior

Install macOS 10.13.3 and Xcode 9.2
Install SuperTux dependencies (cmake, pkg-config, boost, etc) through Mac Homebrew.

```
wget https://github.com/SuperTux/supertux/releases/download/v0.5.1/SuperTux-v0.5.1-Source.tar.gz
tar xf SuperTux-v0.5.1-Source.tar.gz
cd SuperTux-v0.5.1-Source
cmake -DPNG_PNG_INCLUDE_DIR=/usr/local/opt/libpng
make
```

(That special libpng directory is for Homebrew installations.)

##### Additional debugging information

Full build log: https://gist.github.com/apjanke/efb08aa3c3a5691a45f0339d0796a759

Looks like it's caused by this error in the bundled `squirrel` distribution: https://github.com/albertodemichelis/squirrel/issues/98

**Comment by tobbi** [comment]
/cc @maxteufel 

I'm a bit unsure how to proceed here. Updating the squirrel submodule to alberto's master branch fixes the issue, but since SuperTux 0.5.1 is a released version, we cannot retroactively change anything. Does this warrant a bugfix release?

@apjanke Would it be possible to apply https://github.com/albertodemichelis/squirrel/commit/a3a78eec2050162c4912a5208f92b27f658976d8 to the squirrel submodule for the homebrew build in order to get it to compile?

**Comment by apjanke** [comment]
I think so. Homebrew's policy is to generally support patches when they have been accepted by upstream, and since this commit has made it in to `squirrel`'s head, that'd be acceptable. I'll see about applying it.

**Comment by apjanke** [comment]
(Though that'll only fix it for Homebrew users, not all Mac users. I'd personally consider a bugfix release once upstream `squirrel` releases a bugfix release; you probably don't want to be pulling a random version from their `master`.)

**Comment by apjanke** [comment]
Well, I'm afraid that patch doesn't apply cleanly to the last `squirrel` release version; too much intervening change.

The thing to do might be to sit tight and wait for a bugfix or new feature release from `squirrel`, and just update then.

**Comment by maxteufel** [comment]
I don't actually think we rely that much on stable squirrel releases in the submodule (or do we, @tobbi?).

So, I'd suggest cloning the `release/0.5` branch from GitHub, and updating the submodule to whatever latest master HEAD there is. See if that works. If it does, we could probably fix that in the `release/0.5` branch (and possibly backport some other stuff like AppImage support and some bugfixes, then we could release 0.5.2).

**Comment by tobbi** [comment]
I can remember having the same issue and fixing it by updating the squirrel submodule, so it will definitely work.

**Comment by tobbi** [comment]
Tracking with "relnotes" tag so this can be closed once 0.6.0 is released.

**Comment by tobbi** [comment]
Closing for tomorrow's 0.6.0 release.
